src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/WorkloadResolver.cs

```diff
@@ -0,0 +1,212 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.Net.Sdk.WorkloadManifestReader
+{
+    /// <remarks>
+    /// This very specifically exposes only the functionality needed right now by the MSBuild workload resolver
+    /// and by the template engine. More general APIs will be added later.
+    /// </remarks>
+    public class WorkloadResolver : IWorkloadResolver
+    {
+        readonly Dictionary<string, WorkloadDefinition> workloads = new Dictionary<string, WorkloadDefinition>(StringComparer.OrdinalIgnoreCase);
+        readonly Dictionary<string, WorkloadPack> packs = new Dictionary<string, WorkloadPack>(StringComparer.OrdinalIgnoreCase);
+
+        public static WorkloadResolver Create(IWorkloadManifestProvider manifestProvider)
```

Comment:
nit: Factory method is not easy to discover,  any reason it is not a constructor?

Reply:
not really